### PR TITLE
Ignore 3x segment upgrade if unneeded

### PR DESCRIPTION
**Pull request by dakrone**
If the index was created with 0.90.x or later, skip the check for 3x
segments, since Lucene 4.x was used for those versions.

This helps in particular with sharedfs recoveries, where the check for
upgrading segments can cause FS race conditions.

(Note, this is for 1.x only)

Slightly related to #11361

### CODE REVIEW

**Comment by dakrone** [comment]
@s1monw pushed another commit making this encapsulated into the InternalEngine

File: src/main/java/org/elasticsearch/index/engine/InternalEngine.java [hunk 1]
```diff
@@ -118,7 +119,24 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
         boolean success = false;
         try {
             try {
-                upgrade3xSegments(store);
+                String version = engineConfig.getIndexSettings().get("index.version.created");
+                boolean autoUpgrade = true;
```
Comment (s1monw) [inline src/main/java/org/elasticsearch/index/engine/InternalEngine.java]: can you use `Version#indexCreated(Settings)` here instead?

**Comment by s1monw** [comment]
left one comment otherwise LGTM

**Comment by imotov** [comment]
@s1monw Are we saying that v1.6 will not be compatible with indices created before v0.19?

**Comment by s1monw** [comment]
@imotov see https://github.com/elastic/elasticsearch/blob/1.x/src/main/java/org/elasticsearch/gateway/local/state/meta/LocalGatewayMetaState.java#L472 we are upgrading them

**Comment by imotov** [comment]
@s1monw I see. Thanks! It looks like this is going to be a problem only in master, which cannot open these files anyway. It might be a good idea to improve the error message though. I will take a look.

**Pull request merged**